### PR TITLE
Initial commit for schema registry course

### PR DESCRIPTION
**Pull request by bbejeck**
This commit adds the basic infrastructure for the Schema Registry course

### CODE REVIEW

File: schema-registry/build.gradle [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+targetCompatibility = JavaVersion.VERSION_11
+
+repositories {
+    mavenCentral()
+    maven {
+        url = uri("https://packages.confluent.io/maven/")
+    }
+
+    maven {
+        url = uri("https://jitpack.io")
```
Comment (davetroiano) [inline schema-registry/build.gradle:37]: just curious, what dependency is the repo needed for?

File: schema-registry/build.gradle [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+}
+
+
+schemaRegistry {
+    def props = new Properties()
+    file("src/main/resources/confluent.properties").withInputStream {props.load(it)}
+    def srUrl = props.getProperty("schema.registry.url")
+    def fullAuth =  props.getProperty("basic.auth.user.info")
+    if (srUrl != null && fullAuth != null) {
+        //println "Using Schema Registry endpoint:${srUrl}, username:${auth[0]},password:${auth[1]}"
```
Comment (davetroiano) [inline schema-registry/build.gradle:84]: uncomment? or add space after `//`

File: schema-registry/src/main/proto/purchase.proto [hunk 3]
```diff
@@ -0,0 +1,6 @@
+syntax = "proto3";
+
+package io.confluent.developer.proto;
+option java_outer_classname = "PurchaseProto";
+
+//Complete this file by adding the message implementation as directed in the course
```
Comment (davetroiano) [inline schema-registry/src/main/proto/purchase.proto:6]: space after `//`?

File: schema-registry/src/main/resources/confluent.properties.orig [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# This is a placeholder for storing the properties need to connect to Confluent Cloud
```
Comment (davetroiano) [inline schema-registry/src/main/resources/confluent.properties.orig:1]: properties needed

File: schema-registry/src/main/resources/confluent.properties.orig [hunk 5]
```diff
@@ -0,0 +1,3 @@
...
+# Create a new properties file by copying this file to one named confluent.properties
+# The confluent.properties file is ignored by Git - never checkin your credentials!
```
Comment (davetroiano) [inline schema-registry/src/main/resources/confluent.properties.orig:3]: add some placeholders below here with the expected fields and placeholders for credential values?

File: schema-registry/src/main/avro/purchase.avsc [hunk 6]
```diff
@@ -0,0 +1,5 @@
+{
+  "type":"record",
+  "namespace": "io.confluent.developer.avro",
+   // Complete the JSON for the avro record as directed in the course
```
Comment (davetroiano) [inline schema-registry/src/main/avro/purchase.avsc:4]: avro -> Avro

**Review by davetroiano — COMMENTED** [review]
overall LGTM, some minor comments / questions

File: schema-registry/src/main/resources/confluent.properties.orig (hunk 4 above)
Comment (bbejeck) [reply schema-registry/src/main/resources/confluent.properties.orig:1]: this is left blank intentionally - users will rename the file to `confluent.properties` which is ignored.
They can just copy the property file from Confluent Cloud config into here - that's what the instructions will show

Comment (davetroiano) [reply schema-registry/src/main/resources/confluent.properties.orig:1]: oops, this was ambiguous. I meant change "...properties need to connect..." to "... properties needed to connect ..." But we can disregard the comment below about adding placeholders

File: schema-registry/build.gradle (hunk 2 above)
Comment (bbejeck) [reply schema-registry/build.gradle:84]: ack

File: schema-registry/src/main/proto/purchase.proto (hunk 3 above)
Comment (bbejeck) [reply schema-registry/src/main/proto/purchase.proto:6]: ack

File: schema-registry/src/main/avro/purchase.avsc (hunk 6 above)
Comment (bbejeck) [reply schema-registry/src/main/avro/purchase.avsc:4]: ack

File: schema-registry/src/main/resources/confluent.properties.orig (hunk 5 above)
Comment (bbejeck) [reply schema-registry/src/main/resources/confluent.properties.orig:3]: Fair enought

File: schema-registry/src/main/resources/confluent.properties.orig (hunk 4 above)
Comment (bbejeck) [reply schema-registry/src/main/resources/confluent.properties.orig:1]: Ah got it now :)

**Comment by bbejeck** [comment]
Thanks for the review @davetroiano! I've addressed all the comments LMK WYT

**Review by davetroiano — APPROVED** [review]

**Comment by davetroiano** [comment]
LGTM!

**Pull request merged**